sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java

```diff
@@ -30,23 +32,29 @@
  * The result of a {@link BigQueryIO.Write} transform.
  */
 public final class WriteResult implements POutput {
-
   private final Pipeline pipeline;
+  private final TupleTag<TableRow> failedInsertsTag = new TupleTag<>();
+  private final PCollection<TableRow> failedInserts;
```

Comment:
Do you think it makes sense to return not just the rows, but also their errors, as in `PCollection<KV<TableRow, List<ErrorProto>>>` (or perhaps introduce a special type `FailedInsertRow` containing these)?